charts/seed-bootstrap/charts/monitoring/templates/config.yaml

```diff
@@ -68,6 +68,30 @@ data:
         - prometheus-web.garden.svc
       metric_relabel_configs:
 {{ include "prometheus.keep-metrics.metric-relabel-config" .Values.allowedMetrics.cadvisor | indent 6 }}
+
+    - job_name: kube-state-metrics
+      # Service is used, because we only care about metric from one kube-state-metrics instance
+      # and not multiple in HA setup
+      kubernetes_sd_configs:
+      - role: service
+        namespaces:
+          names: [ garden ]
+      relabel_configs:
+      - source_labels: [ __meta_kubernetes_service_label_component ]
+        action: keep
+        regex: kube-state-metrics
+      - source_labels: [ __meta_kubernetes_service_port_name ]
+        action: keep
+      - target_label: instance
+        replacement: kube-state-metrics
+      metric_relabel_configs:
+      - source_labels: [ pod ]
+        regex: ^.+\.tf-pod.+$
+        action: drop
+      - source_labels: [ namespace ]
+        regex: ^shoot--.+$
```

Comment:
The shoot technical ID used to start with `shoot-` (only one hyphen). We should consider this to filter out metrics from namespaces of very old shoots.